lib/langchain/assistant.rb

```diff
@@ -217,6 +217,7 @@ def tool_choice=(new_tool_choice)
     # @return [Array<Langchain::Message>] The messages
     def replace_system_message!(content:)
       messages.delete_if(&:system?)
+      return if content.nil?
```

Comment:
This makes sure we avoid setting a message with nil content, right?

Reply:
Exactly, there is no point to keep message with nil content.